app/validators/form_password_validator.rb

```diff
@@ -4,26 +4,6 @@ module FormPasswordValidator
   included do
     validates :password,
               presence: true,
-              length: Devise.password_length,
-              confirmation: true,
-              if: :password_required?
-  end
-
-  private
-
-  def password_required?
-    new_user? || user_resetting_password? || user_updating_password?
-  end
-
-  def new_user?
-    @user.encrypted_password.blank?
-  end
-
-  def user_resetting_password?
-    @user.reset_password_token.present?
-  end
-
-  def user_updating_password?
-    password.present?
```

Comment:
Yes, but that wasn't the only reason. It was to be able to write a proper test. Since this is related to this change, I think it fits in this PR. Otherwise, I would have to make the test look ugly, then make the test look pretty in the cleanup PR. Might as well take care of both in the same PR since they're tightly coupled.